srv/salt/ceph/macros/README.md

```diff
@@ -0,0 +1,26 @@
+## OS switch macro
+
+Usage:
+
+```
+{% set custom = salt['pillar.get']('your_customization_pillar', 'not a file') %}
+{% from 'ceph/macros/os_switch.sls' import os_switch with context %}
+
+include:
+  - .{{ os_switch(custom) }}
+```
+
+Use this macro to implement an opportunistic operating system switch in your
+state files. The macro will check the directory of the including state files and
+depending on the content output a single state file or directory to include. The
+rules are as follows:
+- Files are preferred directories. However the directory include can be forced
+  by explicitly including the directory's init.sls
+- The following names are tested in order and returned if either file or
+  directory is present in the including sls file's directory.
+- ```custom```
```

Comment:
nitpicky:

maybe use extra indentation here? Technically it's a enumeration within a enumeration:

- The following names are tested in order and returned if either file or￼ directory is present in the including sls file's directory.
  - value of the ```osfinger``` grain, for example ```CentOS```
  - etc..